### PR TITLE
refactor css comments

### PR DESCRIPTION
**Pull request by evilebottnawi**
- [x] I’ve added tests to confirm my change works.
- [ ] (If changing the API or CLI) I’ve documented the changes I’ve made (in the `docs/` directory)
- [x] (If the change is user-facing) I’ve added my changes to `changelog_unreleased/*/pr-XXXX.md` file following `changelog_unreleased/TEMPLATE.md`.
- [x] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/master/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://deploy-preview-7021--prettier.netlify.com/playground/#N4Igxg9gdgLgprEAuEB6VACAglAlgWwEMZdoNyAdKdDAcioDpCNgNIAbCAJyQy7gAmGAL5VGhQkgAWEAG5weSQgDN4XFm2jxYvCiAoH9IANwiMNPVXKUo1jIwBGGh4TABrAOZcIAVygCkHy52AAopGBgAByR0OAAPQnxI9jgGSHwASlNhawYHJCVVBQ1IWAQYXRBCPWyxakwcAmJSW3IAQkYwEohOHn4ha1EoRiFWDm5efvMAKnNMacwcjAYAwrVusp17EHoTETrUWbtrZVOrchpyBeOV6TkFApV14HPrGgBHAHdaBRgAT3srXIpW0FW2tAhNX2QKoAG0pPxlABdO7ydQAMgANHMMABhCD4fDlDr6KDwxEo5QQMA+ADOGCxOPxhPKAAF6GSEXBkUowCR5AyWOcGDAID0SJEhUCMBBIq5cP9eABGYznIbCECYkCykjQWnIUCELjeT4ABSNCH1KEI7E+hD++q1Di4rjccBgAGU5WBcFAPMgYFwfHAteF8OwAOpSBVwWneuAey0K3CyBV-ZDgWmOkC+2m-U0ujxEZDKG15rUAK1pcQAQi73O6PYk4AAZX1wEtlkMgKtxD2+jwpACKPgg8E77HLIDlXDzXAzKSzmunXF9MAjuAEMCkyAAHAAGLWRbx5iMuyIZ4+xhTyZfvUfwAuyq1VWkAWigcEEgmX-HvuH4AtCCLQgJynPN8FwAMg27WkB2HB8OyQUtJ27GBCAcDctx3JAACYtUDQhcHYAdmWLFBYwAVmXOk4AAFQwq0UKnWRgwASX8coPTAVdIhgHABA9f4UjAuBhGEIA)✨**

fixes https://github.com/prettier/prettier/issues/7780

### CODE REVIEW

File: tests/css_less/comments.scss [hunk 1]
```diff
@@ -0,0 +1,12 @@
+// This comment won't be included in the CSS.
```
Comment (thorn0) [inline tests/css_less/comments.scss]: Shouldn't the extension of this file be `.less`?

**Comment by evilebottnawi** [comment]
@thorn0 done :+1:

**Comment by lydell** [comment]
@evilebottnawi The tests seem to be failing

**Comment by evilebottnawi** [comment]
@lydell nothing new in PR) keep in todo for investigation

**Comment by evilebottnawi** [comment]
@lydell @thorn0 trimEnd doesn't supported in node@4 what tips will be to insert a polyfill?

**Comment by thorn0** [comment]
If it's an urgent fix, you can simply use `.replace(/\s+$/, "")` instead. Otherwise, target the 2.0.

**Comment by evilebottnawi** [comment]
Let's do it for next, it is not high priority

File: src/language-css/parser-postcss.js [hunk 2]
```diff
@@ -553,7 +555,10 @@ function requireParser(isSCSSParser) {
     return require("postcss-scss");
   }
 
-  return require("postcss-less");
+  const lessParser = require("postcss-less");
+  return {
+    parse: text => lessParser.parse(replaceQuotesInInlineComments(text))
+  };
```
Comment (evilebottnawi) [inline src/language-css/parser-postcss.js]: Can we add a comment to explain hack?

Comment (thorn0) [reply src/language-css/parser-postcss.js]: done

**Comment by thorn0** [comment]
Ready for review.

Known issue: quotes are stripped from comments in values. Difficult to fix because `value-*` nodes don't have correct location data, only relative to the beginning of the value. I wish `postcss-values-parser` had an option to pass the initial line and column.,,

**Prettier pr-7021**
[Playground link](https://deploy-preview-7021--prettier.netlify.com/playground/#N4Igxg9gdgLgprEAuEBDABMAOldf0QAOqYAljAJ5LoD0N6WIAhIzvvgIwDcOAviABoQRGKWgBnZKFQAnGRADuABVkJJKVABsFqCpKEAjGSQDWcGAGViZKAHNkMGQFc4QgBYwAtpoDqb8nDi1nAWauSkAG7kFMjg4vogpFDicDIwSsa2nqjIAGZaKUIAVuIAHgBCxmBmlqiecAAySXB5Ba4gJaUWSbaacACKThDwrZqFIMQyKTKxffGCEzJJMD6kACYwbsgAHAAMQoTyKT7GhLGHgakRLUIAjkPwGUTqaOIAtFBwcGvfCzJw91I-wyqCyOSQ+TG7RSnlIDmc0J6fUGwxaELaQhgqAMqw2WyQACZMcZSJoegBhCCebKxQIAVgWThSABVsepIeMIi4AJJQH6wCxgJaEGAAQT5FkofVGKV4vCAA)
```sh
--parser less
```

**Input:**
```less
a {
    opacity: // "!"
      1;
}
```

**Output:**
```less
a {
  opacity: //  !
    1;
}

```

**Comment by evilebottnawi** [comment]
@thorn0 sounds good, let's open a issue after merge

**Comment by lydell** [comment]
Would it be possible to replace quotes in comments with some placeholder and then replace back to quotes at the end?

**Comment by thorn0** [comment]
This placeholder has to be one character long. E.g. we could use '\0' for single quotes? But what about double quotes? `\uffff`?

**Comment by lydell** [comment]
Yeah, why not. I think it’s more likely that people put quotes in their comments than `\0` or `\uffff` in their CSS.

**Pull request merged**